### ISSUE TITLE
Bug report - NOS is broken

### ISSUE THREAD

**Issue by BigPimpinBaby** [InZidiuZ/op-framework-issue-tracker#2718]
### Assurance

I have read the `README.md`

### Player Identification

Steam:zeroxium

### Summary

Currently Nos bottles are broken and don’t function in several different ways. Sometimes they fail to appear in your trunk, sometimes you get infinite nos, sometimes you’ll use your entire bottle in one burst, and sometimes it just doesn’t work at all. Also I believe Nos was updated to not damage your engine under a certain temperature, as well as a few other things but I haven’t seen that implemented so idk if that’s related to it being broken 

### Steps to Reproduce

Basically just put Nos into a car and test it. 9/10 times it will scuff, it behaves very weird 

### Expected Behavior

Should work like normal with left shift, shouldn’t damage engine under 90C, I believe it was updated to last a little longer and be a little better as well 

### Actual Results

Nos will drain it one shot, or not drain at all. Screen shaking animation won’t go away. Sometimes won’t appear in your trunk etc 

### Server

International (c3s1.op-framework.com)

**Comment by coalaura** [comment]
Okay i've taken a look at it and did a bunch of bugfixes.

Your engine will ALWAYS be damaged while boosting but you have an additional overheat damage that occurs when your engine temperature is above a certain level.

You can now press Shift+X to purge which will now also decrease your engine temperature slightly (while still using NOS ofc).

Other than that i wasn't able to reproduce any of the bugs you mentioned but like i said i still fixed a few things that COULD potentially break. I tested it out for quite a while and after these fixes personally i felt like NOS was pretty powerful.

Please keep in mind that NOS is and was never intended to be used as a "press button to go fast" thing. You have to time your boosts right and know how to use it, otherwise you will absolutely ruin your engine in lightning speed
